### PR TITLE
Add check for unused function arguments

### PR DESCRIPTION
**Pull request by ZacLN**
#118

### CODE REVIEW

File: src/linting/checks.jl [hunk 1]
```diff
@@ -30,7 +31,9 @@ const LintCodeDescriptions = Dict{LintCodes,String}(IncorrectCallArgs => "Possib
     IncludeLoop => "Loop detected, this file has already been included.",
     MissingFile => "The included file can not be found.",
     InvalidModuleName => "Module name matches that of its parent.",
-    TypePiracy => "An imported function has been extended without using module defined typed arguments.")
+    TypePiracy => "An imported function has been extended without using module defined typed arguments.",
+    UnusedFunctionArgument => "An argument is included in a function signature but not used with its body."
```
Comment (davidanthoff) [inline src/linting/checks.jl:35]:
```suggestion
    UnusedFunctionArgument => "An argument is included in a function signature but not used within its body."
```

**Comment by davidanthoff** [comment]
@ZacLN do you want to merge this? Or do we still need to figure out whether we need to do something else about the new config value? Presumably add an option in the extension itself?

**Comment by ZacLN** [comment]
I've been thinking about how useful this actually is. Surely there are many instances, especially within the context of multiple dispatch, where a function has unused arguments and this doesn't indicate an issue of any sort?

**Comment by davidanthoff** [comment]
Yeah, good point... So maybe we don't do this?

In TypeScript any variable that is not used is shown in the editor in a less bright color (or something like that). So that is a nice visual cue that something is not used anywhere, without showing up in the diagnostic list or as an error. Maybe that could be an option?

**Comment by ZacLN** [comment]
Solution found on the vscode side of things. A little version bump would be needed after merging

**Comment by ZacLN** [comment]
Yep, I think a fair few things are missing from that side of things - I'll have a more detailed look after the config related PRs are merged

**Pull request merged**